src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs

```diff
@@ -144,6 +143,18 @@ public CompilationResult Compile(RelativeFileInfo fileInfo, string compilationCo
 
                     if (!result.Success)
                     {
+                        if (!compilation.References.Any() && !_applicationReferences.Value.Any())
```

Comment:
@davidfowl \ @pakrym  Yeah, this is a guess at best. We could add an API to `DependencyContext` that says it doesn't have compilation context which might be nicer. Thoughts?